### PR TITLE
Changing ifconfig.co to MCR url

### PR DESCRIPTION
**Pull request by jatakiajanvi12**
- Updating the ifconfig.co with the mcr url to avoid curl requests on any external url

Testing steps: 
- Added the domain name to indicate usnat/ussec/azurechinacloud yaml
- Added the MCR_URLs as environment variables in yaml
- Correct URLs are being picked up as given in environment var for non-public clouds
- mcr.microsoft.com is picked up when domain is not provided.

### CODE REVIEW

File: kubernetes/linux/main.sh [hunk 1]
```diff
@@ -334,32 +334,35 @@ if [ -e "/etc/ama-logs-secret/WSID" ]; then
       fi
 
       if [ $? -ne 0 ]; then
+            registry="mcr.microsoft.com"
```
Comment (ganga1980) [inline kubernetes/linux/main.sh:337]: prefix https:// missing.  you can have registry=https://mcr.microsoft.com/v2  and use this value in applicable places

File: kubernetes/linux/main.sh [hunk 2]
```diff
@@ -334,32 +334,35 @@ if [ -e "/etc/ama-logs-secret/WSID" ]; then
...
+            if  [ $CLOUD_ENVIRONMENT == "usnat" ] || [ $CLOUD_ENVIRONMENT == "ussec" ]; then
+                  registry=$MCR_URL
```
Comment (ganga1980) [inline kubernetes/linux/main.sh:339]: Please test the changes.

Comment (ganga1980) [inline kubernetes/linux/main.sh:339]: This requires change in AKS side.

Comment (jatakiajanvi12) [reply kubernetes/linux/main.sh:339]: Not able to create a PR there because in the process of getting permissions. Will merge this PR after that is done

Comment (jatakiajanvi12) [reply kubernetes/linux/main.sh]: Created a PR on aks-rp

Comment (jatakiajanvi12) [reply kubernetes/linux/main.sh]: Tested the changes. Adding the scenarios in the description

File: kubernetes/linux/main.sh [hunk 3]
```diff
@@ -334,32 +352,39 @@ if [ -e "/etc/ama-logs-secret/WSID" ]; then
       fi
 
       if [ $? -ne 0 ]; then
+            registry="https://mcr.microsoft.com/v2/"
+            if [ $CLOUD_ENVIRONMENT == "usnat" ] || [ $CLOUD_ENVIRONMENT == "ussec" ] || [ $CLOUD_ENVIRONMENT == "azurechinacloud" ]; then
+                  if [ ! -z "$MCR_URL" ]; then
```
Comment (ganga1980) [inline kubernetes/linux/main.sh:357]: q1. what will will be the curl response if the MCR_URL is not set?  we should also the log message here if the MCR_URL is empty/not-set.

Comment (jatakiajanvi12) [reply kubernetes/linux/main.sh:357]: Yes makes sense. Added the logic for trying curl only if registry is present. Also, added the log messaged in failure cases

File: kubernetes/linux/main.sh [hunk 4]
```diff
@@ -334,32 +352,43 @@ if [ -e "/etc/ama-logs-secret/WSID" ]; then
...
-            if [ ! -z "$PROXY_ENDPOINT" ]; then
-               if [ -e "/etc/ama-logs-secret/PROXYCERT.crt" ]; then
-                  echo "Making curl request to ifconfig.co with proxy and proxy CA cert"
-                  RET=`curl --max-time 10 -s -o /dev/null -w "%{http_code}" ifconfig.co --proxy $PROXY_ENDPOINT --proxy-cacert /etc/ama-logs-secret/PROXYCERT.crt`
-               else
-                  echo "Making curl request to ifconfig.co with proxy"
-                  RET=`curl --max-time 10 -s -o /dev/null -w "%{http_code}" ifconfig.co --proxy $PROXY_ENDPOINT`
-               fi
+            registry="https://mcr.microsoft.com/v2/"
+            if [ $CLOUD_ENVIRONMENT == "usnat" ] || [ $CLOUD_ENVIRONMENT == "ussec" ] || [ $CLOUD_ENVIRONMENT == "azurechinacloud" ]; then
```
Comment (ganga1980) [inline kubernetes/linux/main.sh:356]: we can have the azurechinacloud in main.sh itself and we dont need to pass down from RP.

Comment (jatakiajanvi12) [reply kubernetes/linux/main.sh]: Updated this.

File: kubernetes/linux/main.sh [hunk 5]
```diff
@@ -334,32 +352,44 @@ if [ -e "/etc/ama-logs-secret/WSID" ]; then
...
+            if [ $CLOUD_ENVIRONMENT == "usnat" ] || [ $CLOUD_ENVIRONMENT == "ussec" ]; then
+                  registry=$MCR_URL
+            fi
+            if [ -z $registry ]; then
+                  echo "The environment variable MCR_URL is not set for CLOUD_ENVIRONMENT: $CLOUD_ENVIRONMENT"
+                  RET=000
             else
-                  echo "Making curl request to ifconfig.co"
-                  RET=$(curl --max-time 10 -s -o /dev/null -w "%{http_code}" ifconfig.co)
+                  if [ ! -z "$PROXY_ENDPOINT" ]; then
```
Comment (ganga1980) [inline kubernetes/linux/main.sh:366]: There is indention issue. Block under if [ ! -z "$PROXY_ENDPOINT" ]; then needs to be intended correctly.

File: kubernetes/linux/main.sh [hunk 6]
```diff
@@ -334,32 +352,44 @@ if [ -e "/etc/ama-logs-secret/WSID" ]; then
...
-                  RET=`curl --max-time 10 -s -o /dev/null -w "%{http_code}" ifconfig.co --proxy $PROXY_ENDPOINT --proxy-cacert /etc/ama-logs-secret/PROXYCERT.crt`
-               else
-                  echo "Making curl request to ifconfig.co with proxy"
-                  RET=`curl --max-time 10 -s -o /dev/null -w "%{http_code}" ifconfig.co --proxy $PROXY_ENDPOINT`
-               fi
+            registry="https://mcr.microsoft.com/v2/"
+            if [ $CLOUD_ENVIRONMENT == "azurechinacloud" ]; then
+                  registry="https://mcr.azk8s.cn/v2/"
+            fi
+            if [ $CLOUD_ENVIRONMENT == "usnat" ] || [ $CLOUD_ENVIRONMENT == "ussec" ]; then
```
Comment (ganga1980) [inline kubernetes/linux/main.sh:359]: I think, we can use **elif** instead of two separate if conditions here.

**Review by ganga1980 — APPROVED** [review]

**Pull request merged**